### PR TITLE
LPS-144034 Test Autom. Filtering the list by clicking on bars in the chart

### PR DESCRIPTION
**Pull request by Luizcarlosqueiroz**
**Ticket**:
[LPS-144034](https://issues.liferay.com/browse/LPS-144034)
[LPS-120676](https://issues.liferay.com/browse/LPS-120676)

**Motivation**:
Test automation: "Filtering the list by clicking on bars in the chart"

### CODE REVIEW

**Comment by Luizcarlosqueiroz** [comment]
Hey @Tim-Cao ! Could you please review this PR?

**Comment by Tim-Cao** [comment]
Just start reviewing

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 1]
```diff
@@ -8,6 +8,10 @@ definition {
 		Click(locator1 = "ContentDashboard#AUDIT_GRAPH_FIRST_BAR");
 	}
 
+	macro clickOnFirstBarColor {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:11]: Refactor ContentDashboard.clickOnFirstBar to
```
ContentDashboard.clickOnBar {
   Click(
            fillColor = "${fillColor}",
            locator1 = "ContentDashboard#BAR_FROM_NO_VOCABULARY_SPECIFIED");
}
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase [hunk 2]
```diff
@@ -35,12 +35,56 @@ definition {
...
+			ContentDashboard.clickOnFirstBarColor(fillColor = "#CDCED9");
+		}
+
+		task ("Apply second filter 'Stage 1' by clicking on first bar") {
+			ContentDashboard.clickOnFirstBar();
+		}
+
+		task ("Assert both filters are applied") {
+			for (var category : list "Audience 1,Stage 1") {
+				AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboardAuditGraph.testcase:77]: Add a new macro in ContentDashboard.macro named viewAppliedFilters 
```
for (var filterLabel : list "${filterLabelList}") {
{
  AssertElementPresent {
		filterLabel = "${filterLabel }",
		locator1 = "ContentDashboard#FILTER_LABEL");
  }
}
```

```
<tr>
	<td>FILTER_LABEL</td>
	<td>//li[contains(@class,'tbar')]//span[contains(@class,'label-item-expand')][starts-with(.,'${filterLabel}')]</td>
	<td></td>
</tr>
```

**Review by Tim-Cao — COMMENTED** [review]
Hi @Luizcarlosqueiroz, could you update according to comments? Thanks

**Comment by Luizcarlosqueiroz** [comment]
Hello @Tim-Cao ! Changes done :)

**Comment by Tim-Cao** [comment]
Would you mind integrating **ContentDashboard.clickOnFirstBar** and **ContentDashboard.clickOnFirstBarColor** ?
Make it reusable then rename then update usages
```
macro clickOnBar {
		Click(
			fillColor = "${fillColor}",
			locator1 = "ContentDashboard#BAR_FROM_NO_VOCABULARY_SPECIFIED");
}
```

**Comment by Luizcarlosqueiroz** [comment]
Hey @Tim-Cao !
I integrated the two macros, but I thought it was better to keep the name as `clickOnFirstBar`, because as this macro uses a path based on the color of the bar, it will always click (or perform any action) on the first one. I think that `clickOnBar` name should be use a future macro that clicks on a specific bar, based on its category. What do you think?

**Comment by Tim-Cao** [comment]
I agree with you @Luizcarlosqueiroz. Good job

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path [hunk 3]
```diff
@@ -194,6 +189,11 @@
 	<td>//*[contains(@class, 'card-body')]//*[contains(@class, 'text-truncate') and contains (text(), '${extensionName}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>FILTER_LABEL</td>
+	<td>//li[contains(@class,'tbar')]//span[contains(@class,'label-item-expand')][starts-with(.,'Category:${filterLabel}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path:194]:
```suggestion
	<td>//li[contains(@class,'tbar')]//span[contains(@class,'label-item-expand')][starts-with(.,'${filterLabel}')]</td>
```

Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path:194]: There is not only category filter but also other types filter

**Review by Tim-Cao — COMMENTED** [review]
Hi @Luizcarlosqueiroz, would you mind updating according to comments? Thanks

**Comment by Tim-Cao** [comment]
Btw @Luizcarlosqueiroz, could you fix SF issue?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path (hunk 3 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path]: In this case, as I was using `starts-with`, the category text was needed. But I changed to `//li[contains(@class,'tbar')]//span[contains(@class,'label-item-expand') and contains (text(), '${filterLabel}')]` , making it reusable.

**Comment by Tim-Cao** [comment]
See changes @Luizcarlosqueiroz

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**